pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -973,7 +973,7 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListenerForPortOrUDS(l
 			rdsName = fmt.Sprintf("%d", pluginParams.Port.Port)
 		}
 		httpOpts := &httpListenerOpts{
-			useRemoteAddress: false,
+			useRemoteAddress: true,
```

Comment:
You might want to add a comment here explaining why the use remote address needs to be set to true (especially the piece about passing app specific timeouts via headers).